homeassistant/components/dynalite/bridge.py

```diff
@@ -0,0 +1,118 @@
+"""Code to handle a Dynalite bridge."""
+
+from dynalite_devices_lib import DynaliteDevices
+from dynalite_lib import CONF_ALL
+
+from homeassistant.const import CONF_HOST
+from homeassistant.core import callback
+
+from .const import DATA_CONFIGS, DOMAIN, LOGGER
+from .light import DynaliteLight
+
+
+class BridgeError(Exception):
+    """Class to throw exceptions from DynaliteBridge."""
+
+    def __init__(self, message):
+        """Initialize the exception."""
+        super().__init__()
+        self.message = message
+
+
+class DynaliteBridge:
+    """Manages a single Dynalite bridge."""
+
+    def __init__(self, hass, config_entry):
+        """Initialize the system based on host parameter."""
+        self.config_entry = config_entry
+        self.hass = hass
+        self.area = {}
+        self.async_add_entities = None
+        self.waiting_entities = []
+        self.all_entities = {}
+        self.config = None
+        self.host = config_entry.data[CONF_HOST]
+        if self.host not in hass.data[DOMAIN][DATA_CONFIGS]:
+            LOGGER.info("invalid host - %s", self.host)
+            raise BridgeError(f"invalid host - {self.host}")
+        self.config = hass.data[DOMAIN][DATA_CONFIGS][self.host]
+        # Configure the dynalite devices
+        self.dynalite_devices = DynaliteDevices(
+            config=self.config,
+            newDeviceFunc=self.add_devices,
+            updateDeviceFunc=self.update_device,
+        )
+
+    async def async_setup(self, tries=0):
+        """Set up a Dynalite bridge."""
+        # Configure the dynalite devices
+        await self.dynalite_devices.async_setup()
+
+        self.hass.async_create_task(
+            self.hass.config_entries.async_forward_entry_setup(
+                self.config_entry, "light"
+            )
+        )
+
+        return True
+
+    @callback
+    def add_devices(self, devices):
+        """Call when devices should be added to home assistant."""
+        added_entities = []
+
+        for device in devices:
+            if device.category == "light":
+                entity = DynaliteLight(device, self)
+            else:
+                LOGGER.debug("Illegal device category %s", device.category)
+                continue
+            added_entities.append(entity)
+            self.all_entities[entity.unique_id] = entity
+
+        if added_entities:
+            self.add_entities_when_registered(added_entities)
+
+    @callback
+    def update_device(self, device):
+        """Call when a device or all devices should be updated."""
+        if device == CONF_ALL:
+            # This is used to signal connection or disconnection, so all devices may become available or not.
+            if self.dynalite_devices.available:
+                LOGGER.info("Connected to dynalite host")
+            else:
+                LOGGER.info("Disconnected from dynalite host")
+            for uid in self.all_entities:
+                self.all_entities[uid].try_schedule_ha()
```

Comment:
Use our dispatch helper to signal the entity method instead.

See eg the tuya integration for an example how to use the dispatch helper:

https://github.com/home-assistant/home-assistant/blob/454e63b69ee3d74d6d8d5324395794642e35c988/homeassistant/components/tuya/__init__.py#L97-L100

https://github.com/home-assistant/home-assistant/blob/454e63b69ee3d74d6d8d5324395794642e35c988/homeassistant/components/tuya/__init__.py#L122-L127

Reply:
fixed. will be pushed shortly